lib/nerves_hub_link/pub_sub/message.ex

```diff
@@ -0,0 +1,37 @@
+defmodule NervesHubLink.PubSub.Message do
+  defstruct type: nil, topic: nil, event: nil, params: nil, reply: nil, reason: nil
+
+  alias __MODULE__, as: M
+
+  @type t :: %M{
+          type: :msg | :join | :close | :disconnect,
+          topic: String.t(),
+          event: String.t() | nil,
```

Comment:
The other thing here is I don't think we need to confine ourselves to the Phoenix paradigm of pubsub using strings everywhere that aren't easily checked with dialyzer. Or require vague/generic topics and events. Instead, we could be much more structured with events as atoms or structs themselves and much more explicit.

Subscriptions could be to specific events rather than a string like
```elixir
NervesHubLink.PubSub.subscribe(:my_event)

# or
NervesHubLink.PubSub.subscribe(MyEvent)
```

Reply:
If we are only passing through events without looking at them I think following the Phoenix way makes most sense. If we are going to be more explicit in Socket, then making atom events would be better, yeah :)